sway-types/src/span.rs

```diff
@@ -188,6 +188,27 @@ impl Span {
         }
     }
 
+    pub fn trim_quotes(self) -> Span {
```

Comment:
This will not be needed if the provided infrastructure is used (`AttributeArg::get_string`).

Reply:
Having a `get_string_span` is a fine addition, but in this case, we just need an `Ident` so simply using `Ident::new_no_span(name)` seams like the simplest option. Note that neither the span created with `new_no_span` nor the one with the span from the `name` argument will have the same `src` like the original name and the rest of the prefixes. That's why I would go for the simplest version here and just use `new_no_span`.

Reply:
> With the Span::trim_quotes solution, aren't we keeping the src accurate to the original

Yes we do, but that original span is not used anywhere later on. The ABI generation is looking only at the value in the `Ident` and we will never have, e.g., an error consuming the final, changed `CallPath`. Actually, having a consumer that would try to get the span of the whole path anyhow wouldn't work, because the prefixes and the suffix are in different files.

So for this particular case I agree, `new_no_span` is the simplest option.